api_gen.py

```diff
@@ -113,7 +113,11 @@ def __init__(self, text):
             self.err_value = f"<{self.code}>-1"
         elif self.code in ('unsigned int', 'haddr_t', 'hsize_t', 'size_t'):
             self.err_condition = "==0"
-            self.err_value = f"<{self.code}>0"
+            if self.fname == 'H5Dget_storage_size':
+                # Special case: https://github.com/h5py/h5py/issues/1475
+                self.err_value = f"<{self.code}>-1"
```

Comment:
I think the best thing to do for this function is to use the `except?` syntax, which the [Cython docs describe](http://docs.cython.org/en/latest/src/userguide/language_basics.html#error-return-values) say is for:

> If all possible return values are legal and you can’t reserve one entirely for signalling errors

(`hsize_t` is unsigned, so -1 can't be represented. I *guess* -1 will be converted to the largest possible `unsigned long long`, which would be implausible as a size. But I'd rather not rely on that guess when Cython has a simple, documented way to handle this case.)

Reply:
`hsize_t` is declared in `api_types_hdf5.pxd` as `ctypedef long long hsize_t`, which I think is a signed integer type.

Reply:
That may be a mistake. It's unsigned in the HDF5 code: https://bitbucket.hdfgroup.org/projects/HDFFV/repos/hdf5/browse/src/H5public.h?at=refs%2Ftags%2Fhdf5-1_10_6#186

The two lines below it in `api_types_hdf5.pxd` explicitly use `signed long long`, so I think `hsize_t` was meant to be different from those.